dom/src/main/java/org/apache/james/mime4j/message/AbstractHeader.java

```diff
@@ -86,6 +86,17 @@ public void addField(Field field) {
         return Collections.unmodifiableList(fields);
     }
 
+    /**
+     * Gets the fields of this header. The returned map will not be
+     * modifiable.
+     *
+     * @return the map of <code>Field</code> objects indexed by names.
+     */
+    @Override
+    public Map<String, List<Field>> getFieldsAsMap() {
```

Comment:
Is it ordered? Otherwise, Collection may be enough, right?

Reply:
Yes it is ordered (linked list) and I kept the underlying types.